service/resource/endpoint/desired_test.go

```diff
@@ -1 +1,133 @@
 package endpoint
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+func Test_Resource_Endpoint_GetDesiredState(t *testing.T) {
+	testCases := []struct {
+		Obj               interface{}
+		SetupEndpoints    []*v1.Endpoints
+		ExpectedEndpoints []Endpoint
+	}{
+		{
+			Obj: &Endpoint{
+				IP:               "1.1.1.1",
+				ServiceName:      "TestService",
+				ServiceNamespace: "TestNamespace",
+			},
+			ExpectedEndpoints: []Endpoint{
+				Endpoint{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+		},
+		{
+			Obj: &Endpoint{
+				IP:               "1.1.1.1",
+				ServiceName:      "TestService",
+				ServiceNamespace: "TestNamespace",
+			},
+			SetupEndpoints: []*v1.Endpoints{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "TestService",
+						Namespace: "TestNamespace",
+					},
+					Subsets: []v1.EndpointSubset{
+						{
+							Addresses: []v1.EndpointAddress{
+								{
+									IP: "1.1.1.1",
+								},
+							},
+						},
+					},
+				},
+			},
+			ExpectedEndpoints: []Endpoint{
+				Endpoint{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+		},
+		{
+			Obj: &Endpoint{
+				IP:               "1.1.1.1",
+				ServiceName:      "TestService",
+				ServiceNamespace: "TestNamespace",
+			},
+			SetupEndpoints: []*v1.Endpoints{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "TestService",
+						Namespace: "TestNamespace",
+					},
+					Subsets: []v1.EndpointSubset{
```

Comment:
Possibly needs another test with multiple subsets, but I am not sure if that is necessary.

Reply:
I added this test.